rust/crd/src/lib.rs

```diff
@@ -314,7 +337,8 @@ impl SparkApplication {
     }
 
     pub fn recommended_labels(&self) -> BTreeMap<String, String> {
-        let mut ls = labels::build_common_labels_for_all_managed_resources(APP_NAME, &self.name());
+        let mut ls =
+            labels::build_common_labels_for_all_managed_resources(APP_NAME, &self.name_unchecked());
```

Comment:
This entire labeling business is a bit annoying but I wanted to point out that other PRs from this series (upgrade to 0.27.1) are  using the `ObjectLabels` struct as seen [here](https://github.com/stackabletech/superset-operator/pull/297/files#diff-cdcf0ccaad22643d0fa068f78d4de053156eeb3bd50c5ab341d9ebf71956061aR68).